test/e2e/e2ethanos/services.go

```diff
@@ -357,112 +357,92 @@ func (q *QuerierBuilder) collectArgs() ([]string, error) {
 
 func RemoteWriteEndpoint(addr string) string { return fmt.Sprintf("http://%s/api/v1/receive", addr) }
 
-// NewFutureReceiver returns a future receiver that can be initiated. It is useful
-// for obtaining a receiver address for hashring before the receiver is started.
-func NewFutureReceiver(e e2e.Environment, name string) e2e.FutureInstrumentedRunnable {
-	return e2e.NewInstrumentedRunnable(e, fmt.Sprintf("receive-%v", name)).
-		WithPorts(map[string]int{"http": 8080, "grpc": 9091, "remote-write": 8081}, "http").
-		Future()
+type ReceiveBuilder struct {
+	e2e.Linkable
+
+	f e2e.FutureInstrumentedRunnable
+
+	maxExemplars    int
+	ingestion       bool
+	hashringConfigs []receive.HashringConfig
+	replication     int
 }
 
-// NewRoutingAndIngestingReceiverFromFuture creates a Thanos Receive instances from a future service.
-// It is configured both for ingesting samples and routing samples to other receivers.
-func NewRoutingAndIngestingReceiverFromFuture(f e2e.FutureInstrumentedRunnable, replicationFactor int, hashring ...receive.HashringConfig) e2e.InstrumentedRunnable {
-	var localEndpoint string
-	if len(hashring) == 0 {
-		localEndpoint = "0.0.0.0:9091"
-		hashring = []receive.HashringConfig{{Endpoints: []string{localEndpoint}}}
-	} else {
-		localEndpoint = f.InternalEndpoint("grpc")
+func NewReceiveBuilder(e e2e.Environment, name string) *ReceiveBuilder {
```

Comment:
Why no comment for this and all the other new exported functions? 🤔

Reply:
Because I was too lazy - will do (:

Reply:
No methods here have comments actually, so let's do this in another PR.